src/features/fireWeather/slices/authenticationSlice.ts

```diff
@@ -0,0 +1,58 @@
+import { createSlice, PayloadAction } from '@reduxjs/toolkit'
+import { AppThunk } from 'app/store'
+
+interface AuthenticationState {
+  isAuthenticated: boolean
+  error: string | null
+}
+
+export const authenticationInitialState: AuthenticationState = {
+  isAuthenticated: false,
+  error: null
+}
+
+const auth = createSlice({
+  name: 'authentication',
+  initialState: authenticationInitialState,
+  reducers: {
+    getAuthenticationStart(state: AuthenticationState) {
+      state.isAuthenticated = false
+    },
+    getAuthenticationFailed(
+      state: AuthenticationState,
+      action: PayloadAction<string>
+    ) {
+      state.isAuthenticated = false
+      state.error = action.payload
+    },
+    getAuthenticationSuccess(
+      state: AuthenticationState,
+      action: PayloadAction<boolean>
+    ) {
+      state.isAuthenticated = action.payload
+    },
+    resetAuthentication(state: AuthenticationState) {
+      state.isAuthenticated = false
+    }
+  }
+})
+
+export const {
+  getAuthenticationStart,
+  getAuthenticationFailed,
+  getAuthenticationSuccess,
+  resetAuthentication
+} = auth.actions
+
+export default auth.reducer
+
+export const createAuthentication = (
+  authenticated: boolean
+): AppThunk => async dispatch => {
+  try {
+    dispatch(getAuthenticationStart())
+    dispatch(getAuthenticationSuccess(authenticated))
+  } catch (err) {
+    dispatch(getAuthenticationFailed(err.toString()))
+  }
+}
```

Comment:
I don't see any purpose of this action creator because there is no side effect involved. Side effect is actually happening when you run promise in `FireWeatherPage.jsx`

Reply:
@Kyubinhan I placed it thinking we can use it if there were any side effects later so that no need to write this function entirely from scratch. Anyway I have removed it.